### PR TITLE
Feat add uniswapv2_simulate benchmark

### PR DESCRIPTION
**Pull request by 0xqd**
Add benchmark for uniswapv2_simulate function

### CODE REVIEW

File: benches/uniswapv2_simulate.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        address: address!("ddF8390cEd9fAD414b1ca1DD4Fe14F881C2Cfa70"),
+        token_a,
+        token_a_decimals: 18,
+        token_b: address!("fc0d6cf33e38bce7ca7d89c0e292274031b7157a"),
+        token_b_decimals: 18,
+        reserve_0: 281815124409715083245_u128,
+        reserve_1: 631976629342354846935765_u128,
+        fee: 300,
+    };
+    let swap_amount = U256::from(694724330990640303_u128);
```
Comment (0xKitsune) [inline benches/uniswapv2_simulate.rs:20]: Can you generate random values for this input?

File: benches/uniswapv2_simulate.rs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    // Data is populated from UniswapV2Pool at block 14597943
+    let token_a = address!("c02aaa39b223fe8d0a0e5c4f27ead9083c756cc2");
+    let pool = UniswapV2Pool {
+        address: address!("ddF8390cEd9fAD414b1ca1DD4Fe14F881C2Cfa70"),
+        token_a,
+        token_a_decimals: 18,
+        token_b: address!("fc0d6cf33e38bce7ca7d89c0e292274031b7157a"),
+        token_b_decimals: 18,
+        reserve_0: 281815124409715083245_u128,
+        reserve_1: 631976629342354846935765_u128,
```
Comment (0xKitsune) [inline benches/uniswapv2_simulate.rs:17]: Can you fuzz these inputs as well?

**Comment by 0xKitsune** [comment]
Thanks for opening this PR. I left a few minor comments. Once these are fixed, we can merge this in.

**Comment by 0xqd** [comment]
I use the valid swapping value to benchmark the performance of a valid swap. If we randomize the input values, it might create inconsistencies in benchmark output, wdyt?

**Comment by 0xqd** [comment]
I updated the input, I believe we keep the state to be static instead of random to make sure when tuning, we have consistent benchmark results.

**Pull request merged**